doc/tutorials/using_celadon_as_uos.rst

```diff
@@ -0,0 +1,113 @@
+*************
+
+* Ubuntu 16.04 or higher with at least 150G free disk space
+* Intel NUC Kit ( Reference Platforms: :ref:`ACRN supported platforms <hardware>` )
+* BIOS version should be 0057 or later, the ``Device Mode`` option is selected
+  for the internal USB 3.0 port header on the USB category of the Devices tab
+* If you need to use ``adb`` or ``fastboot`` tool with a Type-A to Type-A USB cable
+  for the Celadon User VM, then you must provide a USB dongle (e.g. `Dawson Canyon USB 3.0 female
+  to 10-pin header cable <https://www.gorite.com/dawson-canyon-usb-3-0-female-to-10-pin-header-cable>`_)
+  to connect to the internal USB 3.0 port header on the main board of the Intel NUC system.
+  Refer to the `Technical Product Specification
+  <https://www.intel.com/content/dam/support/us/en/documents/mini-pcs/nuc-kits/NUC7i5DN_TechProdSpec.pdf>`_
+  to identify the USB 3.0 port header on the main board.
+* Two HDMI monitors
+
+Build Celadon from source
+*************************
+
+#. Follow the `Build Celadon from source instructions
+   <https://01.org/projectceladon/documentation/getting_started/build-source>`_
+   to get the Celadon project source code ready.
+
+#. Select Celadon build target::
+
+      $ cd <Celadon project directory>
+      $ source build/envsetup.sh
+      $ lunch cel_apl-userdebug
+
+   .. note:: You can run ``lunch`` with no arguments to manually choose your Celadon build variants.
+
+#. Download these additional patches and apply each one individually with the command::
+
+       $ git apply <patch-filename>
+
+   .. table:: ACRN patch list
+      :widths: auto
+      :name: ACRN patch list
+
+      +--------------------------------------------------------------------+-------------------------------------------+
+      | Patch link                                                         | Description                               |
+      +====================================================================+===========================================+
+      | https://github.com/projectceladon/device-androidia/pull/458        | kernel config: Add the support of ACRN    |
+      +--------------------------------------------------------------------+-------------------------------------------+
+      | https://github.com/projectceladon/device-androidia-mixins/pull/293 | graphic/mesa: Add the support of ACRN     |
+      +--------------------------------------------------------------------+-------------------------------------------+
+      | https://github.com/projectceladon/device-androidia/pull/441        | cel_apl: use ttyS0 instead of ttyUSB0     |
+      +--------------------------------------------------------------------+-------------------------------------------+
+      | https://github.com/projectceladon/device-androidia/pull/439        | Disable trusty and pstore                 |
+      +--------------------------------------------------------------------+-------------------------------------------+
```

Comment:
Thanks for the information, I will try it and see if the patch list should be update. :smile:

Reply:
I was able to boot Trusty Celadon on SOS but the ``acrn-dm`` has to be replaced since the default one is not support Trusty storage.
But so far I do not know exactly how to build such ``acrn-dm``, maybe it should be another tutorial :-)

Reply:
> I was able to boot Trusty Celadon as UOS but the `acrn-dm` has to be replaced since the default one is not support Trusty storage.

Do you have the details behind this? There is no flag to be used when building `acrn-dm`, there is a parameter called `--enable-trusty` to be used when starting the VM. Based on your comment, it sounds like there are patches needed that should also be sent upstream?

Reply:
> Do you have the details behind this? There is no flag to be used when building `acrn-dm`, there is a parameter called `--enable-trusty` to be used when starting the VM. Based on your comment, it sounds like there are patches needed that should also be sent upstream?

Exactly, `--enable-trusty` is not functional with the built-in ``acrn-dm`` (looks like a fake parameter ?). I've got a patch from @YadongQi which is supported for NUC platforms, but it is not flexible enough to send upstream so far... maybe we should talk with @NanlinXie for the next step ?

Reply:
Thanks for the clarification. In this case, I would suggest we merge this PR as it is and track enabling Trusty separately in a later PR. Posting a patch not suitable for upstream is likely to become obsolete rather quickly. How long would it take to refactor the patch so it's suitable to be upstreamed?

Reply:
You may reference Celadon's doc: https://github.com/projectceladon/celadon-documentation/blob/master/source/tutorials/acrn_ovmf/acrn_ovmf_celadon.rst
BTW, the `--enable-trusty` is not a fake param, just need changes to make OVMF support trusty. And the patches are maintained by @peterfang. Peter, do you have any plan to upstream the patches.